x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Max.java

```diff
@@ -50,7 +54,13 @@ public class Max extends AggregateFunction implements ToAggregator, SurrogateExp
                 tag = "docsStatsMaxNestedExpression"
             ) }
     )
-    public Max(Source source, @Param(name = "field", type = { "boolean", "double", "integer", "long", "date", "ip" }) Expression field) {
+    public Max(
+        Source source,
+        @Param(
+            name = "field",
+            type = { "boolean", "double", "integer", "long", "date", "ip", "keyword", "text", "long", "version" }
+        ) Expression field
+    ) {
```

Comment:
Right now, other functions I checked use this same logic (the BytesRef compareTo), and behave the same as the SORT command.
So, I'm not sure adding this here would make much sense. If we want to explain it, I wonder if it would be better at ESQL level, instead of at function level